### PR TITLE
36 Unit tests

### LINKED ISSUE #36: Add unit tests
## Subtasks

- [ ] Add tests to [yegor.yaml](https://github.com/objectionary/normalizer/blob/17dc8cbbba28b045f0c4640080df2251e6717861/eo-phi-normalizer/test/eo/phi/rules/yegor.yaml) so that each rule has at least 3 unit tests

### PR DESCRIPTION
**Pull request by Anatolay**
Closes #36.

This PR adds:
- An option to consider only the first output of normalization step during unit tests
- Several unit tests for rules in `yegor.yaml`

---

## PR-Codex overview
This PR introduces a new `RuleTestOption` type, allowing for test-specific options in the EO Phi rules. It also modifies test cases to include these options for more flexibility.

### Detailed summary
- Added `RuleTestOption` type for test-specific options
- Updated test cases to include `RuleTestOption` for more flexibility

> The following files were skipped due to too many changes: `eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by 0crat** [comment]
@Anatolay It is [not a good idea](https://www.yegor256.com/2014/04/15/github-guidelines.html) to name Git branches the way you named this one: "`unit-tests`". You've earned -10 points. Next time, better give your branch the same name as the number of the ticket that you are solving. In this case, a perfect name, for example, would be "`460`". Your running balance is [+5](https://www.eolang.org/zerocracy/objectionary-vitals.html).

File: eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt [hunk 1]
```diff
@@ -1,132 +1,132 @@
...
-  |  |  **|**  | ``⊥``
-  | //ObjectAction// | -> | ``(`` //[Binding]// ``)``
-  |  |  **|**  | ``.`` //Attribute//
-  | //[ObjectAction]// | -> | **eps**
-  |  |  **|**  | //ObjectAction// //[ObjectAction]//
-
-
-
-%% File generated by the BNF Converter (bnfc 2.9.5).
+The Language Syntax
```
Comment (fizruk) [inline eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt:1]: Why is this file changed?

Comment (fizruk) [reply eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt:1]: Did you change file encoding?

**Review by fizruk — CHANGES_REQUESTED** [review]

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Yaml.hs [hunk 2]
```diff
@@ -93,9 +93,16 @@ data RuleTest = RuleTest
   { name :: String
   , input :: Object
   , output :: [Object]
+  , options :: Maybe [RuleTestOption]
   }
   deriving (Generic, FromJSON, Show)
 
+newtype RuleTestOption = TakeOne {take_one :: Bool}
```
Comment (deemp) [inline eo-phi-normalizer/src/Language/EO/Phi/Rules/Yaml.hs:100]: You should rename the constructor to `RuleTestOption` and `take_one` to `takeOne`.

**Comment by deemp** [comment]
@Anatolay, #36 required 5 unit tests for each rule. Please add more unit tests

File: eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt (hunk 1 above)
Comment (Anatolay) [reply eo-phi-normalizer/src/Language/EO/Phi/Syntax/Doc.txt:1]: This change applied automatically when I pulled and built master

File: eo-phi-normalizer/src/Language/EO/Phi/Rules/Yaml.hs (hunk 2 above)
Comment (Anatolay) [reply eo-phi-normalizer/src/Language/EO/Phi/Rules/Yaml.hs:100]: I assumed there could be more options in the future. I can rename for now, though

**Comment by Anatolay** [comment]
> @Anatolay, #36 required 5 unit tests for each rule. Please add more unit tests

Wasn't it discussed that 2-3 per rule should be enough?

**Comment by deemp** [comment]
It was discussed, but not agreed.

**Comment by deemp** [comment]
@Anatolay, it was agreed to write at least 3 tests

**Comment by fizruk** [comment]
Let's merge this, and follow up with more unit tests if necessary, in a separate PR.

**Comment by 0crat** [comment]
@fizruk Thanks for the review! You've earned +49 points for this: +25 as a basis; +34 for the 340 hits-of-code that you reviewed; -10 for too few (3) comments.  Your running balance is [+189](https://www.eolang.org/zerocracy/objectionary-vitals.html).

**Comment by 0crat** [comment]
@Anatolay Thanks for the contribution! You've earned +23 points for this: +30 as a basis; -7 for too many hits-of-code (340 >= 100). Please, [keep](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) them coming. Your running balance is [-5](https://www.eolang.org/zerocracy/objectionary-vitals.html).

**Comment by 0crat** [comment]
@Anatolay Thanks for your review! While we appreciate your effort, reviewing your own contribution isn't recommended. You've earned +5 points: +25 base, -40 for self-review, +20 minimum adjustment. Remember, external reviews often provide valuable insights. Your current balance is [+23](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep up the good work, and consider inviting others to review your future contributions!

**Comment by 0crat** [comment]
@deemp Hey there! 👋 Great job on the review! You've snagged +42 points for this one: +25 for being an awesome reviewer, +6 for tackling those 340 hits-of-code (that's a lot!), and +11 for your 11 insightful comments. 💬 Your effort really shows, especially since you avoided the -10 point deduction for having less than 6 comments. Keep up the fantastic work! Your running balance is now [+130](https://www.eolang.org/zerocracy/objectionary-vitals.html). Remember, the more you contribute, the more you earn. Looking forward to your next review! 🚀

**Pull request merged**